### PR TITLE
[PW-5530] Remove paywithgoogle and googlepay 

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**

The reason to remove the `googlepay` or `paywithgoogle` is that the for google pay and apple pay we don't return anything related to google or apple since the actual payment is done with card. These variants are not included in the payments response or in the notification. The returned payment method is the cc type for example `visa` or `mc`

**Tested scenarios**

Tested with :
- paywithgoogle
- inspect the notification
- inspect the response
- Google pay token is set as visible false

**Fixed issue**:

### CODE REVIEW

**Review by acampos1916 — COMMENTED** [review]
Also, please add tested scenarios to the PR, thanks!

**Review by Morerice — APPROVED** [review]

File: Gateway/Request/CheckoutDataBuilder.php [hunk 1]
```diff
@@ -185,7 +185,7 @@ public function build(array $buildSubject)
         }
 
         $requestBody['additionalData']['allow3DS2'] = true;
-
+        //$requestBody['additionalData']['RequestedTestAcquirerResponseCode'] = 2;
```
Comment (acampos1916) [inline Gateway/Request/CheckoutDataBuilder.php:188]:
```suggestion
```

Comment (AlexandrosMor) [reply Gateway/Request/CheckoutDataBuilder.php:188]: oh no ! that was from testing amazonpay. Thanks @acampos1916

File: Helper/Vault.php [hunk 2]
```diff
@@ -233,4 +235,14 @@ private function getExtensionAttributes(InfoInterface $payment)
         }
         return $extensionAttributes;
     }
+
+    /**
+     * @param $additionalData
+     * @return bool
+     */
+    private function isGooglePay($additionalData): bool
```
Comment (michaelpaul) [inline Helper/Vault.php:243]: What about moving this method to `\Adyen\Payment\Helper\PaymentMethods` so that we can reuse it?

Comment (AlexandrosMor) [reply Helper/Vault.php:243]: For now it is not used from other class and `\Adyen\Payment\Helper\PaymentMethods` I think is be responsible for  fetching the payment methods and functionality around it

Comment (michaelpaul) [reply Helper/Vault.php:243]: I saw that you also added an `isGooglePay` method in `Model/Billing/Agreement.php`.

Comment (AlexandrosMor) [reply Helper/Vault.php]: Check again ?

Comment (michaelpaul) [reply Helper/Vault.php]: Looks good to me! 👍

**Review by michaelpaul — DISMISSED** [review]

**Review by Morerice — APPROVED** [review]

File: view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js [hunk 3]
```diff
@@ -81,6 +81,7 @@ define(
                 'paypal',
                 'applepay',
                 'paywithgoogle',
+                'googlepay',
```
Comment (AlexandrosMor) [inline view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js:84]:
```suggestion
     
```

File: Helper/Vault.php [hunk 4]
```diff
@@ -147,12 +147,6 @@ private function getVaultPaymentToken($payment, array $additionalData): PaymentT
             );
 
             $paymentToken->setGatewayToken($additionalData[self::RECURRING_DETAIL_REFERENCE]);
-
-            if (strpos($additionalData[self::PAYMENT_METHOD], "paywithgoogle") !== false
-                && !empty($additionalData['paymentMethodVariant'])) {
-                $additionalData[self::PAYMENT_METHOD] = $additionalData['paymentMethodVariant'];
-                $paymentToken->setIsVisible(false);
```
Comment (acampos1916) [inline Helper/Vault.php:154]: Removing this means that payments made with Google Pay will still be stored, just as not visible?
Google Pay supports recurring payments https://docs.adyen.com/payment-methods/google-pay/api-only#recurring-payments

File: view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js [hunk 5]
```diff
@@ -81,6 +81,7 @@ define(
                 'paypal',
                 'applepay',
                 'paywithgoogle',
+     
                 'amazonpay'
```
Comment (acampos1916) [inline view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js:85]: `googlepay` payment method code still needs to show the pay button right?

**Review by acampos1916 — COMMENTED** [review]
Left a few comments.

File: Helper/Vault.php (hunk 4 above)
Comment (AlexandrosMor) [reply Helper/Vault.php:154]: I tested and the google pay  token is already set as visible false

**Pull request closed without merging**